exchange/bitswap/bitswap.go

```diff
@@ -385,7 +404,15 @@ func (bs *Bitswap) wantNewBlocks(ctx context.Context, bkeys []u.Key) {
 			}
 		}(p)
 	}
-	wg.Wait()
+	done := make(chan struct{})
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+	select {
+	case <-done:
+	case <-ctx.Done():
+	}
```

Comment:
I have no idea at all what yamux is doing. the method is `waitForSendErr` which selects on the error channel and the shutdown channel and exits when either even happens...